src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/AbstractSmkRuleOrCheckpointType.kt

```diff
@@ -143,18 +144,17 @@ abstract class AbstractSmkRuleOrCheckpointType<T: SmkRuleOrCheckpoint>(
               and containingFile.virtualFile returns null for that copy
             */
             val virtualFile = elem.containingFile.originalFile.virtualFile
-            val elementFile = File(virtualFile?.presentableUrl ?: containingFileName)
-            val fileContentRootDirectory = File(
-                    ProjectRootManager.getInstance(elem.project)
-                            .fileIndex
-                            .getContentRootForFile(virtualFile)
-                            ?.presentableUrl
-                            ?: elementFile.parent
-            )
-            val displayPath = try {
-                elementFile.toRelativeString(fileContentRootDirectory)
-            } catch (e: IllegalArgumentException) { // thrown by toRelativeString if paths have different roots
-                elementFile.name
+            val displayPath = if (virtualFile == null) {
+                containingFileName
+            } else {
+                val fileContentRootDirectory = ProjectRootManager.getInstance(elem.project)
+                        .fileIndex
+                        .getContentRootForFile(virtualFile)
+                if (fileContentRootDirectory == null) {
+                    containingFileName
+                } else {
+                    VfsUtil.getRelativePath(virtualFile, fileContentRootDirectory)
```

Comment:
156 could return `null` if the virtual file path doesn't belong to file content root directory, let's return something useful, e.g. virtual file path